Sources/Sharing/SharedKeys/AppStorageKey.swift

```diff
@@ -15,168 +15,204 @@
   extension SharedReaderKey {
     /// Creates a shared key that can read and write to a boolean user default.
     ///
-    /// - Parameter key: The key to read and write the value to in the user defaults store.
+    /// - Parameters:
+    ///   - key: The key to read and write the value to in the user defaults store.
+    ///   - suiteName: The domain identifier when using an app suite.
     /// - Returns: A user defaults shared key.
-    public static func appStorage(_ key: String) -> Self
+    public static func appStorage(_ key: String, suiteName: String? = nil) -> Self
```

Comment:
I think we'd want to mirror [Apple's API](https://developer.apple.com/documentation/swiftui/appstorage/init(wrappedvalue:_:store:)) here:

```suggestion
    public static func appStorage(_ key: String, store: UserDefaults? = nil) -> Self
```